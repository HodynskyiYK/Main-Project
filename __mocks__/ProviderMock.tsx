import React from 'react'
import {MemoryRouter} from 'react-router-dom'
import { MovieModalContextProvider } from '../src/context/movie-modal-context'
import { Provider } from 'react-redux'
import { store } from '../src/store'
import { MovieDetailsContextProvider } from '../src/context/movie-details-context'

// @ts-ignore
export const ProvidersMock = ({ children }) => {
    return (
        <Provider store={store}>
            <MovieModalContextProvider>
                <MovieDetailsContextProvider>
                    <MemoryRouter>{children}</MemoryRouter>
                </MovieDetailsContextProvider>
            </MovieModalContextProvider>
        </Provider>
    )
}