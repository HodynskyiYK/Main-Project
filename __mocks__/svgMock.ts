// Mock Svg files
// https://github.com/gregberge/svgr/issues/83#issuecomment-437342861
const SvgMock = "icon-mock"
export const ReactComponent = SvgMock
export default SvgMock