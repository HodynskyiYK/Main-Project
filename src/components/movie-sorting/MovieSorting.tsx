import React, {FC} from 'react'

export const MovieSorting: FC = () => {

    return (
        <div className="movie-sorting">
            <p className="label">Sort by</p>
            <p className="selected-value">Release date</p>
            <ul className="sorting-list">
                <li className="sorting-item">
                    <span className="sorting-value">Release date</span>
                </li>
            </ul>
        </div>
    )
}
