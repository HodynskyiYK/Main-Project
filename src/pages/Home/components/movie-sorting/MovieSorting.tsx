import React, { FC, useState } from 'react'
import styles from './MovieSorting.module.scss'
import { useAction, useTypedSelector } from '../../../../hooks'
import { MOVIE_SORTING_TYPES } from './movieSortingConstants'

export const MovieSorting: FC = () => {
    const [showSortingList, setShowSortingList] = useState<boolean>(false)
    const [activeSorting, setActiveSorting] = useState<string>('Release date')
    const {fetchMovies} = useAction()
    const {filterBy} = useTypedSelector(state => state.movies)

    const toggleSortingList = () => {
        setShowSortingList(!showSortingList)
    }

    const changeSortingType = (sortItem: any) => {
        if (sortItem.name !== activeSorting) {
            fetchMovies(filterBy, sortItem.value)
            setActiveSorting(sortItem.name)
        }
        toggleSortingList()
    }

    return (
        <div className={styles.movieSorting}>
            <p className={styles.label}>Sort by</p>
            <p
                className={styles.selectedValue}
                onClick={toggleSortingList}
            >{activeSorting}</p>
            {
                showSortingList && (
                    <ul className={styles.sortingList}>{
                        MOVIE_SORTING_TYPES.map(item => (
                            <li
                                className={styles.sortingItem}
                                key={item.id}
                            >
                                <span
                                    className={styles.sortingValue}
                                    onClick={() => changeSortingType(item)}
                                >{item.name}</span>
                            </li>
                        ))
                    }</ul>
                )
            }
        </div>
    )
}
