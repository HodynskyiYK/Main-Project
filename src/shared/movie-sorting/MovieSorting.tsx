import React, { FC, useEffect, useState } from 'react'
import styles from './MovieSorting.module.scss'
import { useAction, useTypedSelector } from '../../hooks'
import { MOVIE_SORTING_TYPES } from './movieSortingConstants'
import { IMovieSortingItem } from './movieSortingTypes'

export const MovieSorting: FC = () => {
    const [showSortingList, setShowSortingList] = useState<boolean>(false)
    const [activeSorting, setActiveSorting] = useState<string>('Release date')
    const {fetchMovies} = useAction()
    const {filterBy, sortBy} = useTypedSelector(state => state.movies)

    useEffect(() => {
        const activeSortBy = MOVIE_SORTING_TYPES.find((item) => item.value === sortBy)
        if (activeSortBy?.name) {
            setActiveSorting(activeSortBy.name)
        }
    }, [sortBy])

    const toggleSortingList = () => {
        setShowSortingList(!showSortingList)
    }

    const changeSortingType = (sortItem: IMovieSortingItem) => {
        if (sortItem.name !== activeSorting) {
            fetchMovies(filterBy, sortItem.value)
            setActiveSorting(sortItem.name)
        }
        toggleSortingList()
    }

    return (
        <div className={styles.movieSorting}>
            <p className={styles.label}>Sort by</p>
            <p
                className={styles.selectedValue}
                onClick={toggleSortingList}
            >{activeSorting}</p>
            {
                showSortingList && (
                    <ul className={styles.sortingList}>{
                        MOVIE_SORTING_TYPES.map(item => (
                            <li
                                className={styles.sortingItem}
                                key={item.id}
                            >
                                <div
                                    className={styles.sortingValue}
                                    onClick={() => changeSortingType(item)}
                                >
                                    {item.name}
                                </div>
                            </li>
                        ))
                    }</ul>
                )
            }
        </div>
    )
}
