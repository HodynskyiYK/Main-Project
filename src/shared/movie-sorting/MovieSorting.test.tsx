import React from 'react'
import { render } from '@testing-library/react'
import { MovieSorting } from './MovieSorting'
import { ProvidersMock } from '../../../__mocks__/ProviderMock'

describe('MovieSorting tests', () => {

    test('To match snapshot', () => {
        const {asFragment} = render(<ProvidersMock>
            <MovieSorting/>
        </ProvidersMock>)
        expect(asFragment()).toMatchSnapshot()
    })
})