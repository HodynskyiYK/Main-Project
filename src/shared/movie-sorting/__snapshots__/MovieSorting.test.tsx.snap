// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MovieSorting tests To match snapshot 1`] = `
<DocumentFragment>
  <div
    class="movieSorting"
  >
    <p
      class="label"
    >
      Sort by
    </p>
    <p
      class="selectedValue"
    >
      Release date
    </p>
  </div>
</DocumentFragment>
`;
